src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8622,7 +8625,23 @@ internal static List<AlternateStreamData> GetStreams(string path)
             SafeFindHandle handle = NativeMethods.FindFirstStreamW(
                 path, NativeMethods.StreamInfoLevels.FindStreamInfoStandard,
                 findStreamData, 0);
-            if (handle.IsInvalid) throw new Win32Exception();
+            if (handle.IsInvalid)
+            {
+                // Except for a test for handle.IsInvalid, this must be immediately after a P/Invoke call.
+                int error = Marshal.GetLastWin32Error();
+                // Directories don't normally have alternate streams, so this is not an exceptional state.
+                // If a directory has no alternate data streams, FindFirstStreamW returns ERROR_HANDLE_EOF.
+                if (error == NativeMethods.ERROR_HANDLE_EOF && System.IO.Directory.Exists(path))
```

Comment:
Why do we need to check the directory existence?

Reply:
question was raised in https://github.com/PowerShell/PowerShell/issues/10570#issuecomment-694562285 and I didn't get a response, was wondering if it made sense myself.  Thoughts?

Reply:
Absent of data stream is not error for files and directories.

Reply:
If a file doesn't have a data stream, it's in violation of the contract NTFS has (all files have an unnamed stream of type $DATA, which can be canonically addressed as filename.ext::$DATA).  That said, there are other filesystems (many of which are implemented via Dokany) which claim alternate data stream capability, but which have different semantics.  I think I'll take out the Directory.Exists check.

Reply:
I believe we should address only "exact" scenario `Get-Item -Stream "exact_name"`.

Reply:
The "exact" scenario is already handled above (if the requested stream name has wildcards in it, it will not report the stream not being found, but if the requested stream name does not have wildcards in it it will).